scipy/optimize/tests/test_bracket.py

```diff
@@ -286,15 +286,387 @@ def f(x):
+            (None, 1.95, None, 0.3, None, (0.3, 0.0)),
+            (None, 3.25, None, None, 99.4, (99.4, 0)),
+            (None, 4.5, None, -26.3, None, (-26.3, 0)),
+        )
+    )
+    def test_minima_at_boundary_point(self, xl0, xm0, xr0, xmin, xmax, args):
+        f = self.init_f()
+        kwargs = self.get_kwargs(xr0=xr0, xmin=xmin, xmax=xmax, args=args)
+        result = _bracket_minimum(f, xm0, **kwargs)
+        assert result.status == -1
+        assert args[0] in (result.xl, result.xr)
+        assert result.nfev == f.count
+
+    @pytest.mark.parametrize(
+        "xm0",
+        (
+            np.array([[0.55], [0.58]]),
+            np.array([[0.55, 0.56], [0.57, 0.58]]),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "xl0",
+        (
+            0.2,
+            np.array([[0.2], [0.3]]),
+            np.array([[0.2, 0.3],
+                      [0.3, 0.4]])
+        ),
+    )
+    @pytest.mark.parametrize(
+        "xr0",
+        (
+            0.6,
+            np.array([[0.6], [0.8]]),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "xmin",
+        (
+            np.array([[-np.inf], [-1.0]]),
+            np.array([[-np.inf, -1.0], [-2.0, -3.0]]),
+        )
+    )
+    @pytest.mark.parametrize(
+        "xmax",
+        (
+            np.array([[np.inf], [2.0]]),
+            np.array([[np.inf, 2.0], [3.5, 4.5]]),
+        )
+    )
+    @pytest.mark.parametrize(
+        "args",
+        (
+            (0.0, 0.0),
+            (np.array([[0.0], [-0.1]]), np.array([[0.0], [2.5]])),
+        )
+    )
+    def test_vectorized(self, xl0, xm0, xr0, xmin, xmax, args):
+        f = self.init_f()
+        kwargs = self.get_kwargs(xl0=xl0, xr0=xr0, xmin=xmin, xmax=xmax, args=args)
+        result = _bracket_minimum(f, xm0, **kwargs)
+        self.assert_valid_bracket(result)
+        result.nfev == f.count
+
+    def test_special_cases(self):
+        # Test edge cases and other special cases.
+
+        # Test that integers are not passed to `f`
+        # (otherwise this would overflow)
+        def f(x):
+            assert np.issubdtype(x.dtype, np.floating)
+            return x ** 98 - 1
+
+        result = _bracket_minimum(f, -7, xr0=5)
+        assert result.success
+
+        # Test maxiter = 0. Should do nothing to bracket.
+        def f(x):
+            return x**2 - 10
+
+        xl0, xm0, xr0 = -3, -1, 2
+        result = _bracket_minimum(f, xm0, xl0=xl0, xr0=xr0, maxiter=0)
+        assert_equal([result.xl, result.xm, result.xr], [xl0, xm0, xr0])
+
+        # Test scalar `args` (not in tuple)
+        def f(x, c):
+            return c*x**2 - 1
+
+        result = _bracket_minimum(f, -1, args=3)
+        assert result.success
+        assert_allclose(result.fl, f(result.xl, 3))
+
+        # Initial bracket is valid.
+        f = self.init_f()
+        result = _bracket_minimum(f, -0.2, xl0=-1.0, xr0=1.0, args=(0, 0))
+        assert f.count == 3
```

Comment:
In `TestBracketRoot.test_special_cases`, I probably should have checked that the `result` has the correct `x`s and `f`s, too. Let's do that for completeness.

It looks like the other special cases, like when the minimum is at or beyond a limit or endpoint, is are tested elsewhere. Can you think of any edge cases that are specific to `_bracket_minimum`, though?